### PR TITLE
[HttpKernel] Make session-related services extra-lazy

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/EventListener/SessionListener.php [hunk 1]
```diff
@@ -22,13 +20,6 @@
...
-    private $container;
-
-    public function __construct(ContainerInterface $container)
-    {
-        $this->container = $container;
-    }
-
     protected function getSession()
     {
         if (!$this->container->has('session')) {
```
Comment (chalasr) [inline src/Symfony/Component/HttpKernel/EventListener/SessionListener.php]: `$this->container` can be null now

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpKernel/EventListener/SessionListener.php]: good catch, fixed

**Review by chalasr — APPROVED** [review]
closures could have some typehints/return types

File: src/Symfony/Component/HttpKernel/EventListener/SessionListener.php [hunk 2]
```diff
@@ -22,8 +22,6 @@
  */
 class SessionListener extends AbstractSessionListener
 {
-    private $container;
-
```
Comment (fabpot) [inline src/Symfony/Component/HttpKernel/EventListener/SessionListener.php]: Why is it removed?

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpKernel/EventListener/SessionListener.php]: moved to the parent

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**